packages/react/README.md

```diff
@@ -15,3 +15,155 @@
 # @solana/react
 
 This package contains React hooks for building Solana apps.
+
+## Hooks
+
+### `useSignMessage(uiWalletAccount)`
+
+Given a `UiWalletAccount`, this hook returns a function you can call to sign a byte array.
+
+#### Arguments
+
+A config object with the following properties:
+
+-   `message`: A `Uint8Array` of bytes to sign
+
+#### Returns
+
+An object with the following properties:
+
+-   `signature`: The 64-byte Ed25519 signature as a `Uint8Array`
+-   `signedMessage`: The `Uint8Array` of bytes signed by the wallet. These bytes might differ from the input bytes if the wallet modified the message
+
+#### Example
+
+```tsx
+import { useSignMessage } from '@solana/react';
+
+function SignMessageButton({ account, messageBytes }) {
+    const signMessage = useSignMessage(account);
+    return (
+        <button
+            onClick={async () => {
+                try {
+                    const { signature } = await signMessage({
+                        message: messageBytes,
+                    });
+                    window.alert(`Signature bytes: ${signature.toString()}`);
+                } catch (e) {
+                    console.error('Failed to sign message', e);
+                }
+            }}
+        >
+            Sign Message
+        </button>
+    );
+}
+```
+
+> [!NOTE]
+> There exists a plural version of this hook &ndash; `useSignMessages` &ndash; that accepts multiple inputs and returns an array of outputs.
+
+### `useSignTransaction(uiWalletAccount, chain)`
+
+Given a `UiWalletAccount` and a chain that begins with `solana:`, this hook returns a function you can call to sign a serialized transaction.
+
+#### Arguments
+
+A config object with the following properties:
+
+-   `options`: An object with the following properties that govern how the wallet should simulate the transaction for the benefit of the user
+    -   `minContextSlot`: The minimum slot at which to simulate the transaction
+    -   `preflightCommitment`: The commitment at which to simulate the transaction
+-   `transaction`: A `Uint8Array` of bytes that conforms to the [Solana transaction schema](https://solana.com/docs/core/transactions#transaction)
```

Comment:
We should eliminate `preflightCommitment` entirely, as it's not used by wallets (and shouldn't be). They handle their own simulation, and they aren't sending it anyway.

Reply:
OK, I'll nuke the whole `options` object. If `preflightCommitment` isn't relevant, neither is `minContextSlot`.

Reply:
`minContextSlot` is probably still relevant -- I think it can be used to locate the valid blockhash (assuming the wallet uses it). When the wallet receives a blockhash in a tx from the app, it doesn't know if it's valid, and there's no good API for determining this. But it could theoretically fetch the latest slot from the RPC and know if the `minContextSlot` will be satisfied?

Reply:
So IIUC the threat here is that the application happened to fetch the very newest blockhash at slot X, but the wallet's RPC is somehow behind the application at slot X-1 and doesn't know about that blockhash. Because of this the wallet shows the ‘you waited too long to send this transaction’ warning.

If providing `minContextSlot` is important in this context, then isn't `preflightCommitment` also important? Both come into play when deciding if a blockhash is valid.

Also, I literally don't understand why any of this matters in the context of `signTransaction`.

* If I'm a wallet, and
* I'm not being asked to send a transaction, just sign it, and
* It's not a nonce transaction

Why do I care if the blockhash contained within is valid? I should just simulate it with a valid blockhash, sign the thing, and give it back, right?

Reply:
The blockhash is important for determining what chain you're on since we don't have this otherwise in the tx, so if it's provided by the app, the wallet wants to know as much as it can about how it got there. Realistically most apps aren't passing this right now and wallets aren't using it, but I think we have an opportunity to make that the default in The New Way. Come to think of it, where's the optional chain ID we expect?

Reply:
Oh I see; this is that feature where if your wallet is ‘on devnet’ and the app sends you a transaction with a blockhash that's valid on mainnet, then the wallet can be like ‘hey hey whoa, this blockhash is valid on mainnet – if you sign this then someone could land it on main.’

Reply:
> Come to think of it, where's the optional chain ID we expect?

This API asks for the chain in render.

```ts
const doTheThing = useTheThing(account, chain);
```

That way, if the account doesn't support that chain, you find that out in render, rather than later on when you try to sign something.